warehouse/utils/db/types.py

```diff
@@ -11,7 +11,12 @@
 # limitations under the License.
 import datetime
 
-from sqlalchemy import DateTime, TypeDecorator
+from typing import Annotated
+
+from sqlalchemy import Boolean, DateTime, TypeDecorator, sql
+from sqlalchemy.orm import mapped_column
+
+bool_false = Annotated[bool, mapped_column(Boolean, server_default=sql.false())]
```

Comment:
Add some docs, remove the unneeded `Boolean`

```suggestion
# Custom Types for SQLAlchemy Columns `Mapped` values
bool_false = Annotated[bool, mapped_column(server_default=sql.false())]
```

Reply:
https://github.com/pypi/warehouse/pull/14270/commits/1b48842f7b91ca352b1eb351d110723d669db5cb